API/Validators/CustomerValidator.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Data.Entities;
+using FluentValidation;
+using FluentValidation.Validators;
+using System;
+
+namespace Data.Validators
+{
+    internal class CustomerValidator : AbstractValidator<Customer>
+    {
+        public CustomerValidator()
+        {
+            RuleFor(customer => customer.FullName)
+                .NotEmpty()
+                .MinimumLength(5);
+
+            RuleFor(customer => customer.Email)
+                .NotEmpty()
+                .Equal(costumer => costumer.EmailConfirmation)
+                .WithMessage("Emails don't match")
+                .EmailAddress(EmailValidationMode.AspNetCoreCompatible)
+                .WithMessage("Email is not valid");
+
+            RuleFor(customer => customer.Cpf)
+                .NotEmpty()
+                .Must(BeValidCpf)
+                .WithMessage("Cpf is not valid.");
+
+            RuleFor(customer => customer.Cellphone)
+                .NotEmpty()
+                .MinimumLength(8);
+
+            RuleFor(customer => customer.DateOfBirth)
+                .NotEmpty()
+                .WithMessage("Date of birth is not valid.");
+
+            RuleFor(customer => customer.DateOfBirth.AddYears(18))
+                .LessThan(DateTime.Now.Date)
+                .WithMessage("Age must be over 18.");
+
+            RuleFor(customer => customer.EmailSms)
+                .NotEmpty();
+
+            RuleFor(customer => customer.Whatsapp)
+                .NotEmpty();
```

Comment:
```suggestion
                .NotNull();
```
Quando o valor é false o validator age como se não tivesse sido informado.